packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -84,56 +89,107 @@ export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
         @inject(RawProcessFactory) protected readonly rawProcessFactory: RawProcessFactory,
     ) { }
 
+    private get globSubDirectory(): string {
+        return RipgrepSearchInWorkspaceServer.GLOB_SUB_DIRECTORY_PATTERN;
+    }
+
+    private get globCommandArgument(): string {
+        return RipgrepSearchInWorkspaceServer.GLOB_COMMAND_ARGUMENT;
+    }
+
     setClient(client: SearchInWorkspaceClient | undefined): void {
         this.client = client;
     }
 
-    protected getArgs(options?: SearchInWorkspaceOptions): string[] {
-        const args = ['--hidden', '--json'];
-        args.push(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
+    protected getArgs(rootPaths: string[], options?: SearchInWorkspaceOptions): string[] {
+        const args = new Set<string>();
+
+        const appendGlobArgs = (rawPatterns: string[], exclude: boolean) => {
+            rootPaths.forEach(root => {
+                for (const rawPattern of rawPatterns) {
+                    if (rawPattern !== '') {
+                        const globArguments = this.pathToGlobs(rawPattern, root, exclude);
+                        globArguments.forEach(arg => args.add(arg));
+                    }
+                }
+            });
+        };
+
+        args.add('--hidden');
+        args.add('--json');
+
+        args.add(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
         if (options && options.includeIgnored) {
-            args.push('--no-ignore');
+            args.add('--no-ignore');
         }
         if (options && options.maxFileSize) {
-            args.push('--max-filesize=' + options.maxFileSize.trim());
+            args.add('--max-filesize=' + options.maxFileSize.trim());
         } else {
-            args.push('--max-filesize=20M');
+            args.add('--max-filesize=20M');
         }
+
         if (options && options.include) {
-            for (const include of options.include) {
-                if (include !== '') {
-                    args.push('--glob=**/' + include);
-                }
-            }
+            appendGlobArgs(options.include, false);
         }
+
         if (options && options.exclude) {
-            for (const exclude of options.exclude) {
-                if (exclude !== '') {
-                    args.push('--glob=!**/' + exclude);
-                }
-            }
+            appendGlobArgs(options.exclude, true);
         }
+
         if (options && options.useRegExp || options && options.matchWholeWord) {
-            args.push('--regexp');
+            args.add('--regexp');
         } else {
-            args.push('--fixed-strings');
-            args.push('--');
+            args.add('--fixed-strings');
+            args.add('--');
         }
-        return args;
+
+        return Array.from(args);
     }
 
+    private pathToGlobs = (inputPath: string, rootFolder: string, exclude: boolean): string[] => {
```

Comment:
I think the standard method syntax is fine, any reason we do it this way?

Reply:
you are right, thie method used to be a nested function and then decided to move it out,
so it's a leftoever issue from the refactoring. Thanks !